app_feup/lib/controller/load_static/terms_and_conditions.dart

```diff
@@ -7,3 +11,28 @@ Future<String> readTermsAndConditions() async {
     return 'Could not load terms and conditions. Please try again later';
   }
 }
+
+Future<bool> doTermsAndConditionsNeedAcceptance() async {
+  final hash = await AppSharedPreferences.getTermsAndConditionHash();
+  final acceptance = await AppSharedPreferences.areTermsAndConditionsAccepted();
+  final termsAndConditions = await readTermsAndConditions();
+  final currentHash = md5.convert(utf8.encode(termsAndConditions)).toString();
+  if (hash == null) {
+    await AppSharedPreferences.setTermsAndConditionHash(currentHash);
+    return true;
+  }
+
+  if (currentHash != hash) {
+    await AppSharedPreferences.setTermsAndConditionsAcceptance(false);
```

Comment:
I think this is sort of a side effect that should not be in a getter-like method. What do you think?

Reply:
maybe the name should change? this is not supposed to feel getter-like

Reply:
I believe so, maybe `updateAcceptancePreference` or something that makes clear that that preference is getting updated if it needs to